pkg/proxy/ipvs/proxier.go

```diff
@@ -1688,7 +1765,15 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 		curEndpoints.Insert(des.String())
 	}
 
-	for _, epInfo := range proxier.endpointsMap[svcPortName] {
+	endpoints := proxier.endpointsMap[svcPortName]
+
+	if !onlyNodeLocalEndpoints && utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyAwareServiceRouting) {
+		proxier.mu.Lock()
```

Comment:
I think this would deadlock actually? Cause `syncEndpoint` gets the proxy lock `proxy.mu.Lock()` but `syncProxyRule` also retreives the lock at the beginning and ends up calling `syncEndpoint`.